src/test/java/com/google/sps/scheduler/SchedulerTest.java

```diff
@@ -532,19 +605,139 @@ public void scheduleInRangeWhenStartDateEqualsEndDate() {
     calendarEvents.add(eventA);
     calendarEvents.add(eventB);
 
-    List<Task> actualScheduledTasks = scheduleInRange(calendarEvents, sampleTasks, ZURICH_TIME_ZONE, day, day);
-    List<Task> expectedScheduledTasks = Arrays.asList(
-        createTaskWithDue(
+    Scheduler scheduler = new Scheduler(calendarEvents, defaultDurationSample, ZURICH_TIME_ZONE);
+    List<ExtendedTask> actualScheduledTasks = scheduler.scheduleInRange(day, day);
+    List<ExtendedTask> expectedScheduledTasks = Arrays.asList(
+        createDefaultDurationTaskWithDue(
             createDateTime(day, 10, 0, ZURICH_TIME_ZONE)),
-        createTaskWithDue(
+        createDefaultDurationTaskWithDue(
             createDateTime(day, 10, 30, ZURICH_TIME_ZONE)),
-        createTaskWithDue(
+        createDefaultDurationTaskWithDue(
             createDateTime(day, 13, 0, ZURICH_TIME_ZONE)),
-        createTaskWithDue(
+        createDefaultDurationTaskWithDue(
             createDateTime(day, 13, 30, ZURICH_TIME_ZONE)),
-        createTaskWithDue(
+        createDefaultDurationTaskWithDue(
             createDateTime(day, 14, 0, ZURICH_TIME_ZONE)));
 
     Assert.assertEquals(expectedScheduledTasks, actualScheduledTasks);
   }
+
+  @Test
+  public void fitSingleWorkhourslong() {
+    // Events :
+    // Day    : |-------------------------------|
+    // Tasks  : |-------------------------------|
+    List<Event> calendarEvents = Collections.emptyList();
+    LocalDate day = LocalDate.of(2048, 6, 23);
+
+    Scheduler scheduler = new Scheduler(calendarEvents, varyingDurationSample, ZURICH_TIME_ZONE);
+    List<ExtendedTask> actualScheduledTasks = scheduler.scheduleInRange(day, day);
+    List<ExtendedTask> expectedScheduledTasks = Arrays.asList(
+        createCustomDurationTaskWithDue(createDateTime(day, Scheduler.START_HOUR, Scheduler.START_MINUTE, ZURICH_TIME_ZONE), WORKING_HOURS)
+    );
+
+    Assert.assertEquals(expectedScheduledTasks, actualScheduledTasks);
+  }
+
+  @Test
+  public void varyingDurations() {
+    // Events :    |-A|  |-B--|               |-C---|
+    // Day    : |----------------------------------------|
+    // Tasks  : |5m|  |1h|    |-2h-|30m|10m|5m|     |-2h-|
```

Comment:
There is no other 5m scheduled.

Reply:
Ohh thanks. The reason for this is that originially the 5m was not there in the beginning of the day, but I thought it would be a good idea to test for that as well.